File: backend/app_tuyen_sinh_django/utils/enums/auth_type.py
from enum import Enum, auto


class AuthType(Enum):
    EMAIL = auto()
    GOOGLE = auto()
    MICROSOFT = auto()

    # def __init__(self) -> None:
    #     __class__.range_val = range(0, len(__class__))
    #     super().__init__()

    @property
    def auth_type(self):
        return self.name.capitalize()

    @staticmethod
    def from_id(input_id: int):
        return next(x for x in __class__ if x.value == int(input_id))

File: app_tuyen_sinh_flutter/lib/utils/enums/auth_type.dart
import 'package:app_tuyen_sinh_flutter/utils/extensions/enum_extension.dart';
import 'package:app_tuyen_sinh_flutter/utils/mixins/enum_id_mixin.dart';

enum AuthType with EnumIdMixin {
  email(1, 'Email'),
  google(2, 'Google'),
  mircosoft(3, 'Microsoft');

  const AuthType(this.id, this.name);

  @override
  final int id;
  @override
  final String name;

  static AuthType fromId(id) {
    return AuthType.values.fromId(id) as AuthType;
  }
}
